src/main/java/frc/trigon/robot/utilities/SpeedMechanism2d.java

```diff
@@ -0,0 +1,114 @@
+
+        this.targetVelocityLigament = root.append(new MechanismLigament2d("TargetVelocityLigament", 0, 0, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityTopArrowLigament = targetVelocityLigament.append(new MechanismLigament2d("TargetVelocityTopArrowLigament", 0.2 * maxDisplayableVelocity, zeroTopAngle, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityBottomArrowLigament = targetVelocityLigament.append(new MechanismLigament2d("TargetVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kGray)));
+    }
+
+    public void setVelocity(double velocity) {
+        setVelocity(velocity, lastTargetVelocity);
+    }
+
+    public void setVelocity(double velocity, double targetVelocity) {
+        setTargetVelocity(targetVelocity);
+        setArrowAngle(velocity, targetVelocity);
+        currentVelocityLigament.setLength(velocity);
+        setCurrentLigamentColor(velocityToColor(velocity));
+        Logger.recordOutput(key, mechanism);
+    }
+
+    public void setTargetVelocity(double targetVelocity) {
+        lastTargetVelocity = targetVelocity;
+        targetVelocityLigament.setLength(targetVelocity);
+    }
+
+    private void setCurrentLigamentColor(Color8Bit color) {
+        currentVelocityLigament.setColor(color);
+        currentVelocityTopArrowLigament.setColor(color);
+        currentVelocityBottomArrowLigament.setColor(color);
+    }
+
+    private Color8Bit velocityToColor(double velocity) {
+        if (velocity > deadband)
+            return green;
+        else if (velocity < -deadband)
+            return red;
+        return blue;
+    }
+
+    private void setArrowAngle(double velocity, double targetVelocity) {
+        if (velocity > deadband) {
+            currentVelocityTopArrowLigament.setAngle(positiveTopAngle);
+            currentVelocityBottomArrowLigament.setAngle(positiveBottomAngle);
+        } else if (velocity < -deadband) {
+            currentVelocityTopArrowLigament.setAngle(negativeTopAngle);
+            currentVelocityBottomArrowLigament.setAngle(negativeBottomAngle);
+        } else {
+            currentVelocityTopArrowLigament.setAngle(zeroTopAngle);
+            currentVelocityBottomArrowLigament.setAngle(zeroBottomAngle);
+        }
+        if (targetVelocity > deadband) {
+            targetVelocityTopArrowLigament.setAngle(positiveTopAngle);
+            targetVelocityBottomArrowLigament.setAngle(positiveBottomAngle);
+        } else if (targetVelocity < -deadband) {
+            targetVelocityTopArrowLigament.setAngle(negativeTopAngle);
+            targetVelocityBottomArrowLigament.setAngle(negativeBottomAngle);
+        } else {
+            targetVelocityTopArrowLigament.setAngle(zeroTopAngle);
+            targetVelocityBottomArrowLigament.setAngle(zeroBottomAngle);
+        }
```

Comment:
This is not what I meant... You should create a method that accepts the ligaments and the velocity and applies the angles. Not one method for both

Reply:
why should I accept the ligaments? I already have them as a field.

Reply:
To not have repeated code. 
This code for the target ligament and the current ligament is almost identical. The only difference is the ligaments and the velocity. So we extract it into one method in order to not have repeated code.